src/data_info/util/validators.clj

```diff
@@ -172,6 +172,13 @@
              :path path})))
 
 
+(defn stat-is-dir
+  [{:keys [path type]}]
+  (when-not (= :dir type)
+    (throw+ {:error_code error/ERR_NOT_A_FOLDER
+             :path path})))
```

Comment:
I could have called `path-is-dir` instead of adding this function, but making another call to iRODS when I already had the information that I needed seemed like a waste.

Reply:
Yeah, definitely. I've been trying, over time, to remove more and more of that sort of thing from clj-jargon anyway, so definitely better if you just check without making new calls! :+1: